components/datetime/src/pattern/mod.rs

```diff
@@ -107,8 +122,161 @@ impl From<Vec<PatternItem>> for Pattern {
     }
 }
 
+impl fmt::Display for Pattern {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        let mut string = String::new();
```

Comment:
This was because I didn't know how to use the formatter correctly, as the examples I looked up in the Rust docs don't give examples of using multiple formatter calls, and this is probably my first time implementing a `fmt::Display` trait.

One suggestion I have for this type of feedback is that it's difficult to understand the intent of this kind of comment. I would suggest taking the "why do you" out of the feedback, and instead forming it more of a neutral direct statement or suggestion, e.g. "Pushing onto an intermediate string is less efficient than `formatter.write_str()` here, is this intentional?".

Reply:
I didn't know if there was a reason to append to a string instead of directly to the formatter, and assumed you have have done so intentionally. I don't consider myself to have a greater sense of Rust style than you do, which is why I asked it as a question instead of making a concrete suggestion. I don't want to specifically ask you to change something that I'm not certain about. :)

Reply:
Ah, thanks for the clarification. I tend to try to be as explicit as possible on review feedback, as I often find reviews to be sources of contention (especially working on multi-cultural teams.) I didn't interpret the comment as a real question, but more a challenge to the approach. For cases like this I usually rely on [conventional comments](https://conventionalcomments.org/). My old team used them and they were really helpful to know what kind of feedback we were receiving from each other.

Anyway, aside from the meta-review conversation. I rewrote this to write directly into the formatter, and read up a bit on things. I think it's a better solution than creating the string.